### DISCUSSION TITLE
Pin typescript version to a package's version of typescript

### DISCUSSION THREAD

**Discussion by sasial-dev** [renovatebot/renovate#16500] [Request Help]
I've currently got the following config:
```json
{
    "extends": [
      "github>sasial-dev/aftman-renovate-config"
    ],
    "packageRules": [
        {
            "matchPackageNames": ["@rbxts/compiler-types"],
            "versioning": "regex:v?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(-types)?\\.(?<build>\\d+)?$"
        },
        {
            "groupName": "roblox-ts",
            "matchPackageNames": ["typescript", "roblox-ts", "@rbxts/compiler-types"]
        }
    ]
}
```
I'd like `typescript`, to be pinnned to the version of `typescript` in `roblox-ts`'s dependencies (due to limtations with deduping & the library itself that cannot be avoided). Is that possible through renovate?

**Comment by PhilipAbed** [comment]
so you have a hierarchy of 
roblox-ts
      |__ typescript (transitive)
typescript (direct)

and you want that the roblox-ts's typescript be the main? 

i believe renovate will pick up the dependencies found as your Direct dependencies, and update them to the latest,
renovate does not have any way to check for transitive dependencies and know what they are.

Maybe you can block updates for typescript(direct) by using  `packageRules` set it manually to be less or equal to the version of the roblox's , then Whenever you get a pull request on roblox update, you look in the lockfile changes and you upgrade that rule manually 
to the typescript version that exists.

**Reply by sasial-dev** [reply]
the hierarchy is correct!
yeah i was hoping to _not_ have to do it manually 🤔

**Reply by rarkins** [reply]
Unfortunately we definitely can't do that, and it's not something which would be elegantly possible

**Reply by sasial-dev** [reply]
what would be the best alternative? as i'd still want to use renovate for my deps...

**Reply by rarkins** [reply]
Do you definitely need to depend directly on TS in your own dependencies, if it's installed by roblox-ts anyway?

And if so, are you sure it matters if they get out of sync. Node.js's dependency resolution has always been designed to allow more than one version of a dependency at a time

**Reply by PhilipAbed** [reply]
@sasial-dev if i were you i would create a cron-job(daily or weekly whatever you choose) that will extract the child version/range of roblox's typescript from the lockfile, and then updates your renovate.json file with the actual version that you want to pin, just update the packageRules object.
